specifyweb/frontend/js_src/lib/components/SchemaConfig/Tables.tsx

```diff
@@ -75,18 +75,33 @@ export function TableList({
 }: {
   readonly cacheKey: CacheKey;
   readonly getAction: (table: SpecifyTable) => string | (() => void);
-  readonly filter?: (table: SpecifyTable) => boolean;
+  readonly filter?: (showHiddenTables: boolean, table: SpecifyTable) => boolean;
   readonly children?: (table: SpecifyTable) => React.ReactNode;
 }): JSX.Element {
   const [showHiddenTables = false, setShowHiddenTables] = useCachedState(
     cacheKey,
     'showHiddenTables'
   );
 
+  const advancedTables = React.useMemo(() => {
+    const allTables = Object.values(genericTables);
+
+    const filterFunction: ((table: SpecifyTable) => boolean) | undefined =
+      filter?.bind(undefined, showHiddenTables) ??
+      (showHiddenTables ? undefined : ({ isSystem }): boolean => !isSystem);
+
+    const filteredTables =
+      typeof filterFunction === 'function'
+        ? allTables.filter(filterFunction)
+        : allTables;
+
+    return filteredTables.sort(sortFunction(({ name }) => name));
+  }, [filter, showHiddenTables]);
+
   const sortedTables = React.useMemo(
     () =>
       Object.values(genericTables)
-        .filter((table) => (filter ? filter(table) : !table.isSystem))
+        .filter((table) => (filter ? filter(false, table) : !table.isSystem))
```

Comment:
hmm, looks like the edits I did in this PR are lost - https://github.com/specify/specify7/pull/4242/files#diff-c703adeb4352c7832286317011ba686112a5f56f7c0ca20f3a493fc2561197d4R86-R97

do you know why?

that change was important for improving typescript type checking performance by several seconds

Reply:
because your changes were applied to the previous code, before we had the feature to only tables with app resources

Reply:
because they were applied to an old code modified with the new feature 'only show tables that have an app resource'

Reply:
sorry, not understanding this.
can you link to the code commit that removed them?
that PR was a fixing a performance issue - are you saying a user-facing feature remove a performance optimization?

Reply:
the changes you made are on `const advancedTables = ...` which are kept in this PR. The `const sortedTables = ...` and `const tablesToDisplay = ...` are from a new code for the feature 'only display tables that have resources ([here](https://github.com/specify/specify7/pull/4108/files)) '.